### ISSUE TITLE
layout and layout-xkb modules do not display current keyboard layout correctly

### ISSUE THREAD

**Issue by somospocos** [tobi-wan-kenobi/bumblebee-status#485]
#### Summary
layout, layout-xkb

#### Description:
I see two different problems related to both modules.

Prerequisites:

Have two keyboard layouts.

I set mine in **/etc/X11/xorg.conf.d/90-keyboard-layout.conf**, it contains the following line inside the **InputClass** section:

    Option "XkbLayout" "us,es"

setxkbmap reports it correctly:

    $ setxkbmap -query | grep layout
    layout:     us,es

I also have the **xkbgroup** Python 3 module as required by layout-xkb. Importing it in a Python3 console does not throw any import errors.
    
Problem 1:

Have current layout set to us.

Start bumblebee-status as described below.

It displays "us us". One for each module. This is good.

Terminate bumblebee-status.

Switch to second layout by hitting your hotkey of choice for example. Start bumblebee-status as described below. It still displays "en en" instead of "es es".

So far I can't figure out why by looking at the source code. Outside of bumblebee-status, I use another command line tool (https://github.com/ierton/xkb-switch) to query for current layout, and it displays it correctly. As well,

    xkbgroup.XKeyboard().group_symbol

displays the current layout correctly when I run it in a Python 3 console after importing the module.

Problem 2:

Have current layout set to us.

Start bumblebee-status as described below.

It displays "us us". One for each module. This is good.

Keep bumblebee-status running and start switching layouts (by using the hotkeys, not by using the functionality of layout module).

bumblebee-status keeps displaying "us us", it never changes.

Judging by analogy with cpu module, I notice that both layout and layout-xkb modules lack an **update()** method inside the **Method** class. My guess is that this is the reason of the bug (but problem 1 has to be addressed first).

#### How to reproduce

Run bumblebee-status with -m layout layout-xkb arguments. Everything is default.

**Comment by tobi-wan-kenobi** [comment]
Hi,

the missing "update()" should be fine, layout-xkb.py, for example calls "self.current_layout" during each iteration.

Which is why I don't get *at all* why bumblebee-status doesn't work, but the Python module itself does :(

**Comment by somospocos** [comment]
I have just found that my initial bug report was not accurate.

I am using Xephyr to test bumblebee-status and it seems like Xephyr does not inherit the global keyboard layout of Xorg. Which means this requires additional manual intervention after starting i3 with bumblebee-status inside Xephyr. So I have started to test by launching a terminal at startup from i3 config and typing "setkxbmap us,es" manually before testing layout modules.

New results:

1. layout plugin

Left and right mouse clicks work - layout gets changed and bumblebee-status widget gets updated. When switching layout outside of bumblebee (I just type "xkb-switch -n" in terminal) - the bumblebee-status widget goes out of sync and doesn't update.

2. layout-xkb plugin

Everything works fine, it was a false alarm for the reason explained above.
